src/storage/sftp/storage.c

```diff
@@ -37,6 +38,199 @@ struct StorageSftp
+        case LIBSSH2_HOSTKEY_TYPE_ECDSA_256:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ECDSA_256;
+            break;
+#endif
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_384
+        case LIBSSH2_HOSTKEY_TYPE_ECDSA_384:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ECDSA_384;
+            break;
+#endif
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ECDSA_521
+        case LIBSSH2_HOSTKEY_TYPE_ECDSA_521:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ECDSA_521;
+            break;
+#endif
+
+#ifdef LIBSSH2_HOSTKEY_TYPE_ED25519
+        case LIBSSH2_HOSTKEY_TYPE_ED25519:
+            knownHostKeyType = LIBSSH2_KNOWNHOST_KEY_ED25519;
+            break;
+#endif
+
+        default:
+            knownHostKeyType = 0;
+            break;
+    }
+
+    FUNCTION_TEST_RETURN(INT, knownHostKeyType);
+}
+
+/***********************************************************************************************************************************
+Return a match failed message based on known host check failure type
+***********************************************************************************************************************************/
+static const char *
+storageSftpKnownHostCheckpFailureMsg(const int rc)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INT, rc);
+    FUNCTION_TEST_END();
+
+    const char *matchFailMsg;
+
+    switch (rc)
+    {
+        case LIBSSH2_KNOWNHOST_CHECK_FAILURE:
+            matchFailMsg = "LIBSSH2_KNOWNHOST_CHECK_FAILURE";
+            break;
+
+        case LIBSSH2_KNOWNHOST_CHECK_NOTFOUND:
+            matchFailMsg = "not found in known hosts files: LIBSSH2_KNOWNHOST_CHECK_NOTFOUND";
+            break;
+
+        case LIBSSH2_KNOWNHOST_CHECK_MISMATCH:
+            matchFailMsg = "mismatch in known hosts files: LIBSSH2_KNOWNHOST_CHECK_MISMATCH";
+            break;
+
+        default:
+            matchFailMsg = "experienced an unknown failure:";
+            break;
+    }
+
+    FUNCTION_TEST_RETURN_CONST(STRINGZ, matchFailMsg);
+}
+
+/***********************************************************************************************************************************
+Rewrite the user's known_hosts file with a new entry
+***********************************************************************************************************************************/
+static void
+storageSftpUpdateKnownHostsFile(StorageSftp *const this, const int hostKeyType, const String *const host, const char *const hostKey,
+        const size_t hostKeyLen)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+        FUNCTION_LOG_PARAM(STORAGE_SFTP, this);
+        FUNCTION_LOG_PARAM(INT, hostKeyType);
+        FUNCTION_LOG_PARAM(STRING, host);
+        FUNCTION_LOG_PARAM(STRINGZ, hostKey);
+        FUNCTION_LOG_PARAM(SIZE, hostKeyLen);
+    FUNCTION_LOG_END();
+
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        LIBSSH2_KNOWNHOSTS *userKnownHostsList;
+        char *libSsh2ErrMsg;
+        int libSsh2ErrMsgLen;
+        int rc;
```

Comment:
Again, declare on assignment with possible.

Reply:
Updates in
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/4fd017c80716d71a7ffff1f25379b9da196f7001
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/6811405f150c8f1095c3da31d952f0ddebc58be1
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/a3a82046feab0a915e0c116f867818b86ae42621